sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxGatewayStoreModel.java

```diff
@@ -344,7 +344,17 @@ private String ensureSlashPrefixed(String path) {
                        if (!(exception instanceof CosmosException)) {
                            // wrap in CosmosException
                            logger.error("Network failure", exception);
-                           dce = BridgeInternal.createCosmosException(request.requestContext.resourcePhysicalAddress, 0, exception);
+
+                           int statusCode = 0;
+                           if (WebExceptionUtility.isNetworkFailure(exception)) {
+                               if (WebExceptionUtility.isReadTimeoutException(exception)) {
+                                   statusCode = HttpConstants.StatusCodes.REQUEST_TIMEOUT;
+                               } else {
+                                   statusCode = HttpConstants.StatusCodes.SERVICE_UNAVAILABLE;
+                               }
+                           }
+
+                           dce = BridgeInternal.createCosmosException(request.requestContext.resourcePhysicalAddress, statusCode, exception);
```

Comment:
there is only one or two place in the RxGatewayStoreModel which 0 as status code is used. Why not change them in the source?

rather than transforming it here?